### ISSUE TITLE
View/Display dataset (Eye icon) is confusing when dataset is not in OK state

### ISSUE THREAD

**Issue by davelopez** [galaxyproject/galaxy#14432]
**Describe the bug**
If you try to display the contents of a dataset while it is still uploading you will get an empty file download instead. Or if the dataset failed you usually get a `File Not Found ().` or something like this.

**Galaxy Version and/or server at which you observed the bug**
Galaxy Version: 22.05
Commit: any

**To Reproduce**
1. Upload any dataset
2. While it is still uploading, click the "eye" icon to view it
3. Observe your browser downloads an empty file

**Expected behavior**
I expect a proper message indicating why the dataset cannot be displayed or simply the icon being disabled until the dataset can be correctly displayed.

**Screenshots**

![display_dataset](https://user-images.githubusercontent.com/46503462/184129232-1b29e8ee-f12c-4e55-9335-6ecb9224df3a.gif)

**Additional context**
This also happens in the legacy history panel so it might have been around a long time

**Comment by mvdbeek** [comment]
Yes, this would be great to fix, and I recently spent some time making sure we get a proper message if the dataset is purged.  That was pretty easy, but the other cases can be more complicated.

For the running state it depends on the job setup (`outputs_to_working_directory`) whether you're able to view datasets as they're being generated, so disabling the button would depend on the job destination configuration ... that's not super clean.

For the errored state Galaxy pre-creates the output dataset (at least for statically defined outputs) but also tries to copy the outputs, so you may or may not have a dataset, but you probably have at least an empty output.

I guess it never makes sense for the eye icon to trigger a download (this also happens for datatypes that don't have any other rendering method defined), that's something we're setting in the backend.

I guess for the queued and uploading state we can disable the eye icon, can't see how we could legitimately have data already.

**Comment by ahmedhamidawan** [comment]
@mvdbeek Just confirming, we only disable the eye icon if the dataset has the `queued` or `uploading` state?

**Comment by mvdbeek** [comment]
Those are safe to disable in a first step, but there's more that can be done.

**Comment by madeline3000** [comment]
Additionally, if a dataset is empty a blank screen is displayed when clicking the "eye" or inspecting the data. tested this recently again on 22.05.

**Comment by qiagu** [comment]
I had idea long before we could do something to visualize the intermediate state for some datatypes in running.

**Comment by ahmedhamidawan** [comment]
I guess this can be closed because we now have much a improved view for datasets not ready yet:

<img width="1334" height="312" alt="Image" src="https://github.com/user-attachments/assets/bcfac8b2-b754-4a46-b4ab-337cea60fd41" />

<img width="1345" height="388" alt="Image" src="https://github.com/user-attachments/assets/8101daa9-0ad1-487e-b8bb-e670b1bcf3e3" />
